bin/annotate_test_failures

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ruby
+#
+# Usage:
+#   annotate_test_failures [junit-file-path]
+#
+require 'rexml/document'
+
+junit_path = ARGV.first || File.join('build', 'results', 'report.junit')
+title = ENV.fetch('BUILDKITE_LABEL', 'Tests')
+
+unless File.exist?(junit_path)
+  puts "JUnit file not found: #{junit_path}."
+  exit 0
+end
+
+file = File.open(junit_path)
+xmldoc = REXML::Document.new(file)
+
+failures = []
+REXML::XPath.each(xmldoc, '//testcase[failure]') do |testcase|
+  failure = testcase.elements['failure']
+  failures.append <<~ENTRY
+  <details><summary><tt>#{testcase['name']}</tt> in <tt>#{testcase['classname']}</tt></summary>
+  
+  #{failure['message']}
+
+  ```
+  #{failure.text}
+  ```
+  </details>
+  ENTRY
+end
+
+if failures.empty?
+  puts "No test failure found. Removing any previous `#{title}` Buildkite annotation."
+  system('buildkite-agent', 'annotation', 'remove', '--context', title)
```

Comment:
Ok I kinda hoped that `remove` would fail silently if there was no annotation with that context — so that if there's one it deletes it but if there isn't it would be a no-op — but apparently that's not the case, as I just got this:

```
  | 2022-06-29 18:34:33 DEBUG  DELETE https://agent.buildkite.com/v3/jobs/0181b09e-4dca-44e4-8610-8b83fdd87da9/annotations/%F0%9F%94%AC%20Unit%20Tests
  | 2022-06-29 18:34:33 DEBUG  ↳ DELETE https://agent.buildkite.com/v3/jobs/0181b09e-4dca-44e4-8610-8b83fdd87da9/annotations/%F0%9F%94%AC%20Unit%20Tests proto=HTTP/1.1 status=410 Δ=181.218772ms
  | 2022-06-29 18:34:33 FATAL  Failed to remove annotation: DELETE https://agent.buildkite.com/v3/jobs/0181b09e-4dca-44e4-8610-8b83fdd87da9/annotations/%F0%9F%94%AC%20Unit%20Tests: 410 Annotation no longer exists
```

Reply:
Addressed in [9c9b755](https://github.com/Automattic/bash-cache-buildkite-plugin/pull/27/commits/9c9b755d001d21bd71a7229eeb14222336b235de).